tests/ignite/metrics/test_fbeta.py

```diff
@@ -37,26 +37,40 @@ def _output_transform(output):
     return output["y_pred"], output["y"]
 
 
+@pytest.fixture
+def precision_and_recall(request, available_device):
+    name = request.param
+    if name == "none":
+        return None, None
+    elif name == "custom":
+        return (
+            Precision(average=False, device=available_device),
+            Recall(average=False, device=available_device),
+        )
+    raise ValueError(f"Unknown param: {name}")
+
+
 @pytest.mark.parametrize(
-    "p, r, average, output_transform",
+    "precision_and_recall, average, output_transform",
     [
-        (None, None, False, None),
-        (None, None, True, None),
-        (None, None, False, _output_transform),
-        (None, None, True, _output_transform),
-        (Precision(average=False), Recall(average=False), False, None),
-        (Precision(average=False), Recall(average=False), True, None),
+        ("none", False, None),
+        ("none", True, None),
+        ("none", False, _output_transform),
+        ("none", True, _output_transform),
+        ("custom", False, None),
+        ("custom", True, None),
```

Comment:
Let's make it a bit simplier:
```suggestion
"precision_cls, recall_cls, average, output_transform",
[
        (None, None, False, None),
        ... (keep previous code) ...
        
        (lamdba device: Precision(average=False, device=device), lamdba device: Recall(average=False, device=device), True, None),

...
p = precision_cls(available_device)
p._device == torch.device(available_device)

r = recall_cls(available_device)
r._device == torch.device(available_device)

```

Reply:
please check the results